### PR TITLE
[5.6][build] Make it easier to cross-compile the stdlib and use it

### PR DESCRIPTION
**Pull request by finagolfin**
Cherrypick of #40633 and #40707
 
__Explanation:__ Cross-compiling the stdlib and using it as an external SDK, ie not within any toolchain, had these installation blemishes. The symlink to the clang headers would assume the stdlib was installed in a Swift toolchain alongside clang so it used a relative path before, but no longer does when building a standalone stdlib. It would also assume that the cross-compiled toolchain needed to be kept separate from a native one so it appended a subdirectory named after the cross-compile target to the install destdir, which I added a flag to disable. 

__Scope:__ This is all build-related and only used for cross-compilation, with most of it disabled by default because of the new build flag.

__SR Issue:__ None

__Risk:__ Low, disabled by default or for common build configurations

__Testing:__ I [apply both pulls on my daily Android CI when cross-compiling the stdlib](https://github.com/buttaface/swift-android-sdk/blob/ca35796ab09634ae780cfbbc787c4adb97888991/.github/workflows/sdks.yml#L115).

__Reviewer:__ @compnerd

### CODE REVIEW

**Comment by buttaface** [comment]
@artemcm, would you run the CI on this?

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by buttaface** [comment]
Ping, @compnerd, can we get this in?

**Review by compnerd — APPROVED** [review]
Seems that this should be safe - it has been on main for a bit.

**Comment by buttaface** [comment]
@airspeedswift, a tiny 5.6 pull that needs approval.

**Comment by buttaface** [comment]
@compnerd, mind just merging this, or do we need someone else's approval at this point?

**Comment by buttaface** [comment]
@airspeedswift, been waiting on approval for this for awhile, can we get this small build pull in?

**Comment by buttaface** [comment]
Closing since I was told [only regressions or bugfixes wIll be accepted into the release branch after the release shipped](https://github.com/apple/swift/pull/42178#issuecomment-1101642859), though I did submit this months prior.

**Pull request closed without merging**
